### ISSUE TITLE
Export BPA results

### ISSUE THREAD

**Issue by MijalisMendiz** [TabularEditor/TabularEditor#1182]
!!! Please make sure you're using the latest version of Tabular Editor 2.X before submitting a bug !!!
!!! For Tabular Editor 3 issues, go here -> https://github.com/TabularEditor/TabularEditor3/issues !!!

**Describe the bug**
I'm using the script provided in the elegantb.com blog post [Export Best Practice Analyzer Results
](https://www.elegantbi.com/post/exportbparesults) and getting error CS1545: language doesn't allow 'TabularEditor.BestPracticeAnatyzer.AnalyzerResult. Rule' 

**To Reproduce**
1. Go to the C# Script section
2. Paste this code 

```csharp
using TabularEditor.BestPracticeAnalyzer;

var bpa = new Analyzer();
bpa.SetModel(Model);

var sb = new System.Text.StringBuilder();
string newline = Environment.NewLine;

sb.Append("RuleCategory" + '\t' + "RuleName" + '\t' + "ObjectName" + '\t' + "ObjectType" + '\t' + "RuleSeverity" + '\t' + "HasFixExpression" + newline);

foreach (var a in bpa.AnalyzeAll().ToList())
{
    sb.Append(a.Rule.Category + '\t' + a.RuleName + '\t' + a.ObjectName + '\t' + a.ObjectType + '\t' + a.Rule.Severity + '\t' + a.CanFix + newline);
}

sb.Output();
```
3. Click on Run script

**Expected behavior**
A popup window with the analysis results which can be copied to the clipboard as explained in the article.
![image](https://github.com/TabularEditor/TabularEditor/assets/120328169/561b74c7-a6cb-4e65-8aef-fa57fb157f95)

**Observed behavior**
An error windows pops-up instead, displaying the commented error

**Application specifics**
 - Tabular Editor 2 version: 2.22.1
 - Semantic Engine used: Power BI Desktop
 - Model Compatibility Level: [e.g. 1400, 1500, 1520, etc.]

**Additional context**

**Comment by nexus150** [comment]
I have the same error with the same version, hopefully it can be solved.

**Comment by otykier** [comment]
Yeah, this is an unfortunate breaking change, caused by us recently moving to init-only properties (which is a newer C# language feature that's not supported by the C# compiler used inside Tabular Editor when compiling C# scripts).

To workaround this, please update the script to the following, i.e. replacing all references to `a.Rule` with `a.get_Rule()`:

```csharp
using TabularEditor.BestPracticeAnalyzer;

var bpa = new Analyzer();
bpa.SetModel(Model);

var sb = new System.Text.StringBuilder();
string newline = Environment.NewLine;

sb.Append("RuleCategory" + '\t' + "RuleName" + '\t' + "ObjectName" + '\t' + "ObjectType" + '\t' + "RuleSeverity" + '\t' + "HasFixExpression" + newline);

foreach (var a in bpa.AnalyzeAll().ToList())
{
    sb.Append(a.get_Rule().Category + '\t' + a.RuleName + '\t' + a.ObjectName + '\t' + a.ObjectType + '\t' + a.get_Rule().Severity + '\t' + a.CanFix + newline);
}

sb.Output();
```

**Comment by MijalisMendiz** [comment]
Thanks for the quick solution. With this change, it works fine!

**Comment by otykier** [comment]
This has now been fixed in [2.23.0](https://github.com/TabularEditor/TabularEditor/releases/tag/2.23.0). Please revert to use your original script when you upgrade to this version.

**Comment by MijalisMendiz** [comment]
Thank you so much Daniel. Cheers!
